packages/core/http/core-http-router-server-internal/src/request.ts

```diff
@@ -222,6 +223,13 @@ export class CoreKibanaRequest<
       options,
     };
   }
+  /** infer route access from path if not declared */
+  private getAccess(request: RawRequest): 'internal' | 'public' {
+    return (
+      ((request.route?.settings as RouteOptions)?.app as KibanaRouteOptions)?.access ??
+      (request.path.startsWith('/internal') ? 'internal' : 'public')
```

Comment:
We can be even stricter and add a trailing slash `/internal/` or use RegExp: `new RegExp('^(\/internal$|\/internal\/)')`

However, a route that starts with `/internal...` will almost certainly be intended for `access: internal` unless we _ever_ have a route `/internalize` or something 😅